CMakeLists.txt

```diff
@@ -0,0 +1,79 @@
+cmake_minimum_required(VERSION 3.10)
+project(sg14 CXX)
+
+find_package(Threads REQUIRED)
+
+set(CMAKE_CXX_STANDARD 17)
```

Comment:
I don't suppose there's any way to communicate to CMake that it should use "17 if possible, or 14 otherwise or if specifically requested, or 11 otherwise or if specifically requested..."?  IMHO all of the SG14 stuff should build in '14 and if it doesn't it's probably a bug. (But it should not *break* when built in '17, either, so ideally we'd have a clean way to test both. I don't think CMake is really set up for that kind of thing though, so that's okay.)

Reply:
I don't think either, but I will look into it. I would consider that a "nice to have", maybe in a follow up PR if I get it to work? I will set 14 as the default. I'll probably have to follow up with fixes for MSVC as well.